src/theory/fp/theory_fp.cpp

```diff
@@ -855,22 +855,26 @@ bool TheoryFp::collectModelValues(TheoryModel* m,
           nm->mkNode(kind::FLOATINGPOINT_COMPONENT_SIGNIFICAND, node);
 
       eq::EqualityEngine* ee = m->getEqualityEngine();
-      Assert(ee->hasTerm(compNaN) && ee->getRepresentative(compNaN).isConst());
-      Assert(ee->hasTerm(compInf) && ee->getRepresentative(compInf).isConst());
-      Assert(ee->hasTerm(compZero)
-             && ee->getRepresentative(compZero).isConst());
+      Assert(ee->hasTerm(compNaN));
+      Assert(ee->hasTerm(compInf));
+      Assert(ee->hasTerm(compZero));
+      Node rCompNaN = ee->getRepresentative(compNaN);
+      Node rCompInf = ee->getRepresentative(compInf);
+      Node rCompZero = ee->getRepresentative(compZero);
```

Comment:
```suggestion
      TNode rCompNaN = ee->getRepresentative(compNaN);
      TNode rCompInf = ee->getRepresentative(compInf);
      TNode rCompZero = ee->getRepresentative(compZero);
```

Reply:
As we are trying to get rid of the overhead of three uint64-additions, we should not add unnecessary reference counting here...

Reply:
Fair.
( FWIW my goal wasn't about overhead as such, more about simplicity of code. )

Reply:
Yeah, the whole block is only executed when assertions are enabled, so the overhead shouldn't really matter :)